structural_imbalance.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2020 D-Wave Systems Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import click
+import matplotlib
+import matplotlib.pyplot as plt
+
+import dwave_networkx as dnx
+
+from neal import SimulatedAnnealingSampler
+from dwave.system import DWaveSampler, EmbeddingComposite, LeapHybridSampler
+
+from drawing import draw_social_network
+from dwave_structural_imbalance_demo.mmp_network import global_signed_social_network
+
+
+@click.command()
+@click.option('--qpu', 'sampler_type', flag_value='qpu',
+              help='Use the QPU')
+@click.option('--cpu', 'sampler_type', flag_value='cpu',
+              help='Use simulated annealing')
+@click.option('--hybrid', 'sampler_type', flag_value='hybrid',
+              help="Use Leap's hybrid sampler")
+@click.option('--region', default='global',
+              type=click.Choice(['global', 'iraq', 'syria'],
+                                case_sensitive=False))
+@click.option('--show', is_flag=True,
+              help="show the plot rather than saving it")
+def main(sampler_type, region, show):
+
+    if sampler_type is None:
+        print("No solver selected, defaulting to hybrid")
+        sampler_type = 'hybrid'
+
+    if region == 'global' and sampler_type == 'qpu':
+        print("Given region is too large for the QPU, please choose another "
+              "region or use hybrid.")
```

Comment:
So when running `python structural_imbalance.py --qpu` which is a likely choice for new users wanting to use the quantum computer, the demo will print "please choose another region or use hybrid" but keep running?
a. Should it not stop so the user can follow this message?
b.  The user used the default region, did not chose a region, so maybe something like, "please chose a smaller region..."  

Why not default to QPU and Syria?

Reply:
Also, this is the only sentence that has a period at the end. Please be consistent in following or violating the rules of grammar.

Reply:
The future is hybrid! I think the reason that this demo is interesting relative to the illustrative demo is because it let's users run on a larger data set. That's why I default it to global/hybrid.